test/e2e_awssharedvpc/aws_shared_vpc_test.go

```diff
@@ -0,0 +1,307 @@
+		t.Fatalf("Failed to create external DNS %q: %v", testExtDNSName, err)
+	}
+	defer func() {
+		_ = kubeClient.Delete(context.TODO(), &extDNS)
+	}()
+
+	// Create a service of type LoadBalancer with the annotation targeted by the ExternalDNS resource.
+	t.Log("Creating source service")
+	service := common.DefaultService(testServiceName, testNamespace)
+	if err := kubeClient.Create(context.TODO(), service); err != nil {
+		t.Fatalf("Failed to create test service %s/%s: %v", testNamespace, testServiceName, err)
+	}
+	defer func() {
+		_ = kubeClient.Delete(context.TODO(), service)
+	}()
+
+	// Get the service address (hostname or IP) and the resolved service IPs of the load balancer.
+	serviceAddress, serviceIPs, err := common.GetServiceIPs(context.TODO(), t, kubeClient, common.DnsPollingTimeout, types.NamespacedName{Name: testServiceName, Namespace: testNamespace})
+	if err != nil {
+		t.Fatalf("failed to get service IPs %s/%s: %v", testNamespace, testServiceName, err)
+	}
+
+	// Query Route 53 API with assume role ARN from the dns config, then compare the results to ensure it matches the
+	// service hostname.
+	t.Logf("Querying Route 53 API to confirm DNS record exists in a different AWS account")
+	expectedHost := fmt.Sprintf("%s.%s", testServiceName, hostedZoneDomain)
+	if err := wait.PollUntilContextTimeout(context.TODO(), common.DnsPollingInterval, common.DnsPollingTimeout, true, func(ctx context.Context) (done bool, err error) {
+		recordValues, err := getDNSRecordValues(hostedZoneID, expectedHost, "A")
+		if err != nil {
+			t.Logf("Failed to get DNS record for shared VPC zone: %v", err)
+			return false, nil
+		} else if len(recordValues) == 0 {
+			t.Logf("No DNS records with name %q", expectedHost)
+			return false, nil
+		}
+
+		if _, found := recordValues[serviceAddress]; !found {
+			if _, foundWithDot := recordValues[serviceAddress+"."]; !foundWithDot {
+				t.Logf("DNS record with name %q didn't contain expected service IP %q", expectedHost, serviceAddress)
+				return false, nil
+			}
+		}
+		t.Logf("DNS record with name %q found in shared Route 53 private zone %q and matched service IPs", expectedHost, hostedZoneID)
+		return true, nil
+	}); err != nil {
+		t.Fatalf("Failed to verify that DNS has been correctly set.")
+	}
+
+	t.Logf("Querying for DNS record inside cluster VPC using a dig pod")
+
+	// Verify that the IPs of the record created by ExternalDNS match the IPs of load balancer obtained in the previous
+	// step. We will start a pod that runs a dig command for the expected hostname and parse the dig output.
+	if err := wait.PollUntilContextTimeout(context.TODO(), common.DnsPollingInterval, common.DnsPollingTimeout, true, func(ctx context.Context) (done bool, err error) {
+		gotIPs, err := common.DnsQueryClusterInternal(ctx, t, kubeClient, kubeClientSet, testNamespace, expectedHost)
+		if err != nil {
+			t.Errorf("Failed to query hostname inside cluster: %v", err)
```

Comment:
Are you referring to the `t.Errorf`? Yea I should switch that to `t.Logf`.

So I am a little confused.  I thought `t.Errorf` isn't supposed to fail the test, but just classify the log as an "error" and keep going.

However, in my testing, if there was a `t.Errorf` logged, the test would keep going, then it would fail at the end since the last log statement was an `t.Errorf`.

Reply:
Yes, `T.Errorf()` continues the test execution but marks the test as failed.